core/src/MegaMolGraph.cpp

```diff
@@ -574,6 +568,12 @@ bool megamol::core::MegaMolGraph::add_module(ModuleInstantiationRequest_t const&
         return false;
     }
 
+    Module::ptr_type module_ptr = module_description->CreateModule(request.id);
```

Comment:
```suggestion
    Module::ptr_type module_ptr = module_description->CreateModule(request.id, module_lifetime_dependencies);
```

I guess the next step is to pass the actual resources through the CreateModule() infrastructure to the module constructors, right?

Reply:
I'm not sure if we want to do this. First, with the current module concept the constructors should be as lightweight as possible and basically do nothing else except slot registration to be dummy constructible by the configurator. I don't know if providing resources in the constructor is needed therefore. Second, it would be a massive breaking change, needing to touch every single module constructor. (Basically the same argument as for the idea of putting the resource request in the constructor).
Therefore, I would currently vote for keepings things as they are, at least until we finished thinking about the whole core refactoring. With making the resource request static we at least removed them from being bound to the module runtime and have the flexibility to inject dependencies into the constructor in future.